### PR TITLE
Add NovoGrad optimizer

### PR DESCRIPTION
**Pull request by DT6A**
NovoGrad optimizer was presented in this paper: https://arxiv.org/pdf/1905.11286.pdf
It was used to train the ASR model named Jasper: https://arxiv.org/pdf/1904.03288.pdf

### CODE REVIEW

**Comment by DT6A** [comment]
> There are a couple of conflicts, could you sync and merge so we get this submitted?

Hello! Yes, I will try to fix as soon as possible

File: docs/api.rst [hunk 1]
```diff
@@ -149,6 +149,7 @@ Gradient Transforms
     scale_by_adam
     scale_by_belief
     scale_by_factored_rms
+    scale_by_novograd
```
Comment (hbq1) [inline docs/api.rst:152]: Could you also add `novograd` to the Common Optimizers section?

File: optax/_src/transform.py [hunk 2]
```diff
@@ -952,6 +952,94 @@ def update_fn(updates, state, params=None):
...
+    b1: decay rate for the exponentially weighted average of grads.
+    b2: decay rate for the exponentially weighted average of squared grads.
+    eps: term added to the denominator to improve numerical stability.
+    eps_root: term added to the denominator inside the square-root to improve
+      numerical stability when backpropagating gradients through the rescaling.
+    weight_decay: a scalar weight decay rate.
+    mu_dtype: optional `dtype` to be used for the first order accumulator; if
+      `None` then the `dtype is inferred from `params` and `updates`.
+  Returns:
+    An (init_fn, update_fn) tuple.
```
Comment (hbq1) [inline optax/_src/transform.py:982]: "A `GradientTransformation` object."

**Review by hbq1 — APPROVED** [review]

**Comment by hbq1** [comment]
Thanks a lot for the contribution!

**Comment by hbq1** [comment]
@DT6A could you please add a test for the new optimizer?

**Comment by DT6A** [comment]
@hbq1 is there any reference how the test should be done?

**Comment by mkunesch** [comment]
Hi! We are in the process of writing instructions on this at the moment - in your case it should be enough to just add the new optimizer to the list of optimizers under test in `alias_test.py`. Let me know if you have any questions on this!

**Comment by DT6A** [comment]
@mkunesch Thanks, added it

**Comment by mkunesch** [comment]
(If there are details of the optimizer that would be important to test in isolation it would be good to add a unit test for them, but from a quick look at the documentation and code nothing stood out to me)

**Review by hbq1 — APPROVED** [review]

**Comment by hbq1** [comment]
Merged! Thanks again for this great contribution 🎉

**Pull request closed without merging**